src/modules/ekf2/EKF/mag_fusion.cpp

```diff
@@ -333,63 +334,24 @@ bool Ekf::fuseYaw(const float innovation, const float variance, estimator_aid_so
 
 bool Ekf::fuseDeclination(float decl_sigma)
 {
-	// assign intermediate state variables
-	const float magN = _state.mag_I(0);
-	const float magE = _state.mag_I(1);
-
-	// minimum North field strength before calculation becomes badly conditioned (T)
-	constexpr float N_field_min = 0.001f;
-
 	// observation variance (rad**2)
 	const float R_DECL = sq(decl_sigma);
 
-	// Calculate intermediate variables
-	if (fabsf(magN) < sq(N_field_min)) {
-		// calculation is badly conditioned close to +-90 deg declination
-		return false;
-	}
+	Vector24f H;
+	float innovation;
+	float innovation_variance;
 
-	const float HK0 = ecl::powf(magN, -2);
-	const float HK1 = HK0*ecl::powf(magE, 2) + 1.0F;
-	const float HK2 = 1.0F/HK1;
-	const float HK3 = 1.0F/magN;
-	const float HK4 = HK2*HK3;
-	const float HK5 = HK3*magE;
-	const float HK6 = HK5*P(16,17) - P(17,17);
-	const float HK7 = ecl::powf(HK1, -2);
-	const float HK8 = HK5*P(16,16) - P(16,17);
-	const float innovation_variance = -HK0*HK6*HK7 + HK7*HK8*magE/ecl::powf(magN, 3) + R_DECL;
-	float HK9;
-
-	if (innovation_variance > R_DECL) {
-		HK9 = HK4/innovation_variance;
-	} else {
+	sym::ComputeMagDeclinationInnovInnovVarAndH(getStateAtFusionHorizonAsVector(), P, getMagDeclination(), R_DECL, FLT_EPSILON, &innovation, &innovation_variance, &H);
```

Comment:
Instead of calling a magic `getMagDeclination()` helper would it be better to do it explicitly so we know what's going on? 

Do we ever want to be in here if mag isn't aligned in flight? Would we ever want to fuse `EKF2_MAG_DECL` or should that just be an optimization for mag heading init?

Reply:
Likewise for `getMagDeclination()` usage in `limitDeclination()`.

Reply:
Only just saw your comment now...